packages/devtools_app/test/http/curl_command_test.dart

```diff
@@ -0,0 +1,284 @@
+          method: 'GET',
+          uri: Uri.parse('https://www.example.com'),
+          headers: {
+            'accept-language': ['en-GB,de-DE'],
+            'authorization': ['Bearer \'this is a\' test']
+          },
+        ),
+      );
+
+      expect(
+        command.toString(),
+        "curl --location --request GET 'https://www.example.com' \\\n--header 'accept-language: en-GB,de-DE' \\\n--header 'authorization: Bearer '\\''this is a'\\'' test'",
+      );
+    });
+
+    test('no line breaks when "multiline" is false', () {
+      final command = CurlCommand.from(
+        _getMockedDartIOHttpRequestData(
+          method: 'POST',
+          uri: Uri.parse('https://www.example.com'),
+          headers: {
+            'accept-language': ['en-GB,de-DE'],
+            'authorization': ['Bearer \'this is a\' test']
+          },
+          requestBody: Uint8List(0),
+        ),
+        multiline: false,
+      );
+
+      expect(
+        command.toString(),
+        "curl --location --request POST 'https://www.example.com' --header 'accept-language: en-GB,de-DE' --header 'authorization: Bearer '\\''this is a'\\'' test' --data-raw ''",
+      );
+    });
+
+    test('no --location when followRedirects is false', () {
+      final command = CurlCommand.from(
+        _getMockedDartIOHttpRequestData(
+          method: 'GET',
+          uri: Uri.parse('https://www.example.com'),
+          headers: {},
+        ),
+        multiline: false,
+        followRedirects: false,
+      );
+
+      expect(
+        command.toString(),
+        "curl --request GET 'https://www.example.com'",
+      );
+    });
+
+    test('parses GET request from test_data', () {
+      final command = CurlCommand.from(httpGet);
+
+      expect(
+        command.toString(),
+        "curl --location --request GET \'https://jsonplaceholder.typicode.com/albums/1\' \\\n--header 'content-length: 0'",
+      );
+    });
+
+    test('parses POST request from test_data', () {
+      final command = CurlCommand.from(httpPost);
+
+      expect(
+        command.toString(),
+        "curl --location --request POST \'https://jsonplaceholder.typicode.com/posts\' \\\n--data-raw '    {\n      title: '\\''foo'\\'',\n      body: '\\''bar'\\'',\n      userId: 1,\n    }\n    '",
+      );
+    });
+  });
+}
+
+class _MockDartIOHttpRequestData extends DartIOHttpRequestData {
+  _MockDartIOHttpRequestData(
+    int timelineMicrosBase,
+    this._mockRequest,
+  ) : super(timelineMicrosBase, _mockRequest);
+
+  final HttpProfileRequest _mockRequest;
+
+  @override
+  String? get requestBody {
+    final body = super.requestBody;
+    if (body != null) {
+      return body;
+    }
+
+    if (_mockRequest.requestBody != null) {
+      return String.fromCharCodes(_mockRequest.requestBody!);
+    }
+
+    return null;
+  }
+
+  @override
+  Future<void> getFullRequestData() async {
+    // Do nothing
+  }
+}
+
+DartIOHttpRequestData _getMockedDartIOHttpRequestData({
```

Comment:
nit: avoid starting method names with get. rename to _mockedDartIOHttpRequestData